src/sys_fs_cgroup.c

```diff
@@ -0,0 +1,1021 @@
+// cgroup globals
+
+static int cgroup_enable_cpuacct_stat = CONFIG_ONDEMAND_ONDEMAND;
+static int cgroup_enable_cpuacct_usage = CONFIG_ONDEMAND_ONDEMAND;
+static int cgroup_enable_memory = CONFIG_ONDEMAND_ONDEMAND;
+static int cgroup_enable_blkio = CONFIG_ONDEMAND_ONDEMAND;
+static int cgroup_enable_new_cgroups_detected_at_runtime = 1;
+static int cgroup_check_for_new_every = 10;
+static char *cgroup_cpuacct_base = NULL;
+static char *cgroup_blkio_base = NULL;
+static char *cgroup_memory_base = NULL;
+
+static int cgroup_root_count = 0;
+static int cgroup_root_max = 50;
+static int cgroup_max_depth = 0;
+
+void read_cgroup_plugin_configuration() {
+	cgroup_check_for_new_every = config_get_number("plugin:cgroups", "check for new plugin every", cgroup_check_for_new_every);
+
+	cgroup_enable_cpuacct_stat = config_get_boolean_ondemand("plugin:cgroups", "enable cpuacct stat", cgroup_enable_cpuacct_stat);
+	cgroup_enable_cpuacct_usage = config_get_boolean_ondemand("plugin:cgroups", "enable cpuacct usage", cgroup_enable_cpuacct_usage);
+	cgroup_enable_memory = config_get_boolean_ondemand("plugin:cgroups", "enable memory", cgroup_enable_memory);
+	cgroup_enable_blkio = config_get_boolean_ondemand("plugin:cgroups", "enable blkio", cgroup_enable_blkio);
+
+	char filename[FILENAME_MAX + 1];
+	snprintf(filename, FILENAME_MAX, "%s%s", global_host_prefix, "/sys/fs/cgroup/cpuacct");
+	cgroup_cpuacct_base = config_get("plugin:cgroups", "path to /sys/fs/cgroup/cpuacct", filename);
+
+	snprintf(filename, FILENAME_MAX, "%s%s", global_host_prefix, "/sys/fs/cgroup/blkio");
+	cgroup_blkio_base = config_get("plugin:cgroups", "path to /sys/fs/cgroup/blkio", filename);
+
+	snprintf(filename, FILENAME_MAX, "%s%s", global_host_prefix, "/sys/fs/cgroup/memory");
+	cgroup_memory_base = config_get("plugin:cgroups", "path to /sys/fs/cgroup/memory", filename);
+
+	cgroup_root_max = config_get_number("plugin:cgroups", "max cgroups to allow", cgroup_root_max);
+	cgroup_max_depth = config_get_number("plugin:cgroups", "max cgroups depth to monitor", cgroup_max_depth);
+
+	cgroup_enable_new_cgroups_detected_at_runtime = config_get_boolean("plugin:cgroups", "enable new cgroups detected at run time", cgroup_enable_new_cgroups_detected_at_runtime);
+}
+
+// ----------------------------------------------------------------------------
+// cgroup objects
+
+struct blkio {
+	int updated;
+
+	char *filename;
+
+	unsigned long long Read;
+	unsigned long long Write;
+	unsigned long long Sync;
+	unsigned long long Async;
+	unsigned long long Total;
+};
+
+// https://www.kernel.org/doc/Documentation/cgroup-v1/memory.txt
+struct memory {
+	int updated;
+
+	char *filename;
+
+	unsigned long long cache;
+	unsigned long long rss;
+	unsigned long long rss_huge;
+	unsigned long long mapped_file;
+	unsigned long long writeback;
+	unsigned long long pgpgin;
+	unsigned long long pgpgout;
+	unsigned long long pgfault;
+	unsigned long long pgmajfault;
+/*	unsigned long long inactive_anon;
```

Comment:
minor ocd: i would put the `/*` on separate line, so when enabling the line, the actual line with code is not changed on that changeset, i.e `git annotate`